pkg/node/raft_store_test.go

```diff
@@ -24,32 +26,34 @@ func Test_StoreInMemOpenSingleNode(t *testing.T) {
 	eciesKey, err := ecies.GenerateKey(rand.Reader, secp256k1.S256(), nil)
 	require.NoError(t, err)
 
-	key := CosignerEd25519Key{
+	key := pcosigner.CosignerEd25519Key{
 		PubKey:       dummyPub,
 		PrivateShard: []byte{},
 		ID:           1,
 	}
 
-	cosigner := NewLocalCosigner(
+	cosigner := pcosigner.NewLocalCosigner(
 		log.NewNopLogger(),
-		&RuntimeConfig{},
-		NewCosignerSecurityECIES(
-			CosignerECIESKey{
+		&pcosigner.RuntimeConfig{},
+		pcosigner.NewCosignerSecurityECIES(
+			pcosigner.CosignerECIESKey{
 				ID:        key.ID,
 				ECIESKey:  eciesKey,
 				ECIESPubs: []*ecies.PublicKey{&eciesKey.PublicKey},
 			}),
 		"",
 	)
 
+	validator := &ThresholdValidator{
+		myCosigner: cosigner}
 	s := &RaftStore{
-		NodeID:      "1",
-		RaftDir:     tmpDir,
-		RaftBind:    "127.0.0.1:0",
-		RaftTimeout: 1 * time.Second,
-		m:           make(map[string]string),
-		logger:      nil,
-		cosigner:    cosigner,
+		NodeID:             "1",
+		RaftDir:            tmpDir,
+		RaftBind:           "127.0.0.1:0",
+		RaftTimeout:        1 * time.Second,
+		m:                  make(map[string]string),
+		logger:             nil,
+		thresholdValidator: validator,
```

Comment:
Why does the `RaftStore` need a reference to the `thresholdValidator`?

Reply:
Because of Type Hierarchies. 

Not the greatest refactoring. Basically I introduced "passing via embedding" of the cosigner into RaftStore via thresholdValidator.cosigner.

So it introduces a nil pointer if its ´nil´ in RaftStore, hence needed to introduce the thresholdvariable in the test.

The underlying cause is actually the high coupling of the GRPC server. Fixing this is also on the list of TODO in the next PR.